smarts/core/agent_interface.py

```diff
@@ -225,6 +238,8 @@ class DoneCriteria:
     """
     agents_alive: Optional[AgentsAliveDoneCriteria] = None
     """If set, triggers the ego agent to be done based on the number of active agents for multi-agent purposes."""
+    actors_alive: Optional[ActorsAliveDoneCriteria] = None
+    """If set, triggers the ego agent to be done based on actors existing in the simulation."""
```

Comment:
If possible, it might be better to combine `agents_alive` and `actors_alive` options together into one event configuration, because separate `agents_alive` and `actors_alive` options might conflict.

Reply:
I think what you are considering is the difference between AND and OR causing termination. It is slightly difficult to satisfy these complex cases without introducing expressions.

I think we do not have the case for AND currently.

Reply:
When we use a single env to cater to two types of scenarios, namely (i) one which uses and terminates with the social agent as the leader, and (ii) one which uses and terminates with the sumo vehicle as the leader, we end up with a code which looks like

[Assume lines 151 to 158 are uncommented]
https://github.com/huawei-noah/SMARTS/blob/7cdc5c3dd5b87b51026cfd4256bb2c79c539e2be/smarts/env/gymnasium/platoon_env.py#L151-L163

Here, the env will terminate immediately every time as one of the done criteria, namely `actors_alive` or `agents_alive` will become true immediately at the start of the every scenario.

Another issue here is the name of vehicle of interest differs in both scenarios, because `social-agent-leader` string gets prefixed to the agent name when using social agent as the leader.